src/libraries/System.Text.Json/src/System/Text/Json/Node/JsonArray.cs

```diff
@@ -11,6 +11,10 @@ namespace System.Text.Json.Node
     /// <summary>
     ///   Represents a mutable JSON array.
     /// </summary>
+    /// <remarks>
+    /// It is safe to perform multiple concurrent read operations on a <see cref="JsonArray"/>,
+    /// but issues can occur if the collection is modified while it's being read.
```

Comment:
> but issues can occur 

From the tests, `InvalidOperationException` is thrown. Should we document that explicitly?

Reply:
This is a subset of `List<T>` doc which doesn't document possible exceptions in the "thread safety" section so we are consistent this way at least. Ideally this exception would be documented in the various enumerators returned from `GetEnumerator()`, but since interfaces (`IEnumerator` and `IEnumerator<>`) are returned it is not always possible.

The exceptions thrown by the underlying List\Dictionary used by the node-based code have a nice message associated with them like "Collection was modified; enumeration operation may not execute" so at least the misuses will be consistent and easy to understand. The new tests that check for `InvalidOperationException` were added because it is possible for the node-based code to not throw this nice message (if the enumeration code was using array indexers for example), and perhaps even work in some cases depending on timing -- so the tests verify consistent behavior with List\Dictionary.